api/python/cellxgene_census/src/cellxgene_census/experimental/_embedding.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2022-2023 Chan Zuckerberg Initiative Foundation
+#
+# Licensed under the MIT License.
+
+"""
+Methods to support simplifed access to community contributed embeddings.
+"""
+from __future__ import annotations
+
+import json
+import warnings
+from typing import Optional, Union
+
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import pyarrow as pa
+import tiledbsoma as soma
+
+from .._open import _build_soma_tiledb_context, open_soma
+from .._release_directory import get_census_version_directory
+
+
+def get_embedding(
+    census_version: str,
+    embedding_uri: str,
+    obs_soma_joinids: Union[npt.NDArray[np.int64], pa.Array],
```

Comment:
Any plans to add support for passing in value filters and/or tiledbsoma.AxisQuery objects?

Reply:
The current logic in the notebook seems quite circular.

 - First, you need to figure out what census version to use, so you load the embedding metadata.
 - Then, you open write out the verbose, nested context managers to get the query and soma join IDs.
 - Finally, you get the slice of the embedding, passing in the census version you retrieved from the embedding metadata.

Why are the first two steps necessary? It seems much cleaner to require only the embedding URI and the AxisQuery. The census version can be inferred from the metadata and the right version of census can be opened to fetch the soma join ids if a value filter was provided in AxisQuery.

Reply:
You are raising multiple issues and I think maybe conflating them.

1. the notebook logic - I get the census version from the embedding, as I choose to make the embedding the only input to the notebook. We could make the user specify both, but i"m not sure why that is better.  Keep in mind that they must be consistent or the results are garbage.

2. the API design - it is a requirement that the sugar API check that the user has not accidentally used an embedding and census which are mismatched.  CC: @pablo-gar as this is his design.

Ultimately, to "get" info, we need:
* an embedding URI
* a census URI
* and they must match (and it is a requirement that we check that they match)

I personally prefer a low-level API, which is more flexible for a variety of use cases. This one provides that by accepting an array of joinids, rather than some higher level abstraction such as an OBS dataframe filter.

We could do both.  Easy peasy to implement.

Alec - can you shortcut this conversation and propose a concrete API signature that you would like to see?